flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/source/enumerator/ContinuousIcebergEnumerator.java

```diff
@@ -115,7 +116,12 @@ private ContinuousEnumerationResult discoverSplits() {
       return new ContinuousEnumerationResult(
           Collections.emptyList(), enumeratorPosition.get(), enumeratorPosition.get());
     } else {
-      return splitPlanner.planSplits(enumeratorPosition.get());
+      AtomicReference<ContinuousEnumerationResult> result = new AtomicReference<>();
+      Tasks.foreach(enumeratorPosition.get())
```

Comment:
I was thinking about more like max allowed consecutive planning failures (not retries), because planning is scheduled periodical already

Reply:
Done

Reply:
I was thinking about sth in the same line as Flink's tolerable checkpoint failures.

https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/config/#execution-checkpointing-tolerable-failed-checkpoints

We don't do any retry here.

Reply:
Changed the code. This is what you had in mind?